dev-tools/ls-apis/src/main.rs

```diff
@@ -0,0 +1,235 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Show information about Progenitor-based APIs
+
+use anyhow::{Context, Result};
+use camino::Utf8PathBuf;
+use clap::{Args, Parser, Subcommand};
+use omicron_ls_apis::{
+    AllApiMetadata, ApiDependencyFilter, LoadArgs, ServerComponentName,
+    SystemApis,
+};
+
+#[derive(Parser)]
+#[command(
+    name = "ls-apis",
+    bin_name = "ls-apis",
+    about = "Show information about Progenitor-based APIs"
+)]
+struct LsApis {
+    /// path to metadata about APIs
+    #[arg(long)]
+    api_manifest: Option<Utf8PathBuf>,
+
+    #[command(subcommand)]
+    cmd: Cmds,
+}
+
+#[derive(Subcommand)]
+enum Cmds {
+    /// print out an Asciidoc table summarizing the APIs
+    Adoc,
+    /// print out each API, what exports it, and what consumes it
+    Apis(ShowDepsArgs),
+    /// print out APIs exported and consumed by each deployment unit
+    DeploymentUnits(DotArgs),
+    /// print out APIs exported and consumed, by server component
+    Servers(DotArgs),
+}
+
+#[derive(Args)]
+pub struct ShowDepsArgs {
+    /// Show the Rust dependency path resulting in the API dependency
+    #[arg(long)]
+    show_deps: bool,
+
+    /// Show only API dependencies matching the filter
+    #[arg(long, default_value_t)]
+    filter: ApiDependencyFilter,
+}
+
+#[derive(Args)]
+pub struct DotArgs {
+    /// Show output that can be fed to graphviz (dot)
+    #[arg(long)]
+    dot: bool,
```

Comment:
nit: would recommend turning this into a `--format human|dot` setting.

Reply:
Done in 5fb81d807ae55132758ab78ca5f9217f2c9c1906.